crates/bevy_ecs/src/query/extensions.rs

```diff
@@ -0,0 +1,69 @@
+use std::fmt::Debug;
+
+use thiserror::Error;
+
+use crate::prelude::Query;
+
+use super::{Fetch, FilterFetch, ReadOnlyFetch, WorldQuery};
+
+impl<'w, Q: WorldQuery, F: WorldQuery> Query<'w, Q, F>
+where
+    F::Fetch: FilterFetch,
+{
+    /// Takes exactly one result from the query. If there are no results, or more than 1 result, this will return an error instead.
+    pub fn get_unique(&self) -> Result<<Q::Fetch as Fetch<'_>>::Item, UniqueQueryError<'_, Q>>
```

Comment:
i dont really consider this to be an "extension". I see nothing about it that distinguishes it from other Query methods like get/iter, so imo it should also live alongside the other impls.

 (and in general i think calling things "extensions", "helpers", etc is overly generic).